Packages/ClientRuntime/Sources/Networking/Http/CRT/SDKDefaultIO.swift

```diff
@@ -9,12 +9,22 @@ import AwsCommonRuntimeKit
 import class Foundation.ProcessInfo
 
 public final class SDKDefaultIO {
-    public static let shared = SDKDefaultIO()
+    static weak var privateShared: SDKDefaultIO? = nil
     
-    public let eventLoopGroup: EventLoopGroup
-    public let hostResolver: DefaultHostResolver
-    public let clientBootstrap: ClientBootstrap
-    public let tlsContext: TlsContext
+    public static var shared: SDKDefaultIO {
+        if let shared = privateShared {
+            return shared
+        } else {
+            let shared = SDKDefaultIO()
+            privateShared = shared
+            return shared
+        }
+    }
```

Comment:
Is this how we are suppose to do singletons? Seems like back in the day of obj-c we'd use dispatch_once ?